api/src/handlers/hours.ts

```diff
@@ -68,3 +65,69 @@ export const checkIn = async (
   }
 };
 
+/**
+ * Update a student's hours based on their student number, add the transaction and update their hours.
+ *
+ * @param event The APIGatewayProxyEvent for the API.
+ * @returns The updated user object.
+ */
+
+// TODO: decrease hours?
+export const updateHours = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  if (!event.body) {
+    return ResponseUtilities.createErrorResponse(
+      ErrorConstants.VALIDATION_BODY_MISSING
+    );
+  }
+
+  const data = JSON.parse(event.body);
+
+  if (!data.studentNumber) {
+    return ResponseUtilities.createErrorResponse(
+      ErrorConstants.VALIDATION_BODY_STUDENTNUMBER
+    );
+  }
+  if (!data.checkIn) {
+    return ResponseUtilities.createErrorResponse(
+      ErrorConstants.VALIDATION_BODY_CHECKIN
+    );
+  }
+  if (!data.checkOut) {
+    return ResponseUtilities.createErrorResponse(
+      ErrorConstants.VALIDATION_BODY_CHECKOUT
+    );
+  }
+
+  const params = {
+    TableName: process.env.userTable,
+    IndexName: "StudentNumberIndex",
+    KeyConditionExpression: "studentNumber = :v_title",
+    ExpressionAttributeValues: {
+      ":v_title": data.studentNumber,
+    },
+    ScanIndexForward: false,
+  };
+  try {
+    const userList = await DynamoUtilities.query(params, dynamoDb);
+
+    if (userList.length != 1) {
+      throw new Error(ErrorConstants.DYNAMO_NONUNIQUE_STUDENTNUMBER);
+    }
```

Comment:
I'm thinking that we could move this logic to the dynamo wrapper instead, and make the `query` more specific like `queryUser`. It's going to be reused anywhere we do a query based on student number.

Reply:
did you do this?

Reply:
nah